x/sdk/p2p/client.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/common"
+	"github.com/ava-labs/avalanchego/utils/sampler"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var (
+	ErrAppRequestFailed = errors.New("app request failed")
+	ErrRequestPending   = errors.New("request pending")
+)
+
+type AppResponseCallback func(
+	nodeID ids.NodeID,
+	responseBytes []byte,
+	err error,
+)
+type CrossChainAppResponseCallback func(
+	chainID ids.ID,
+	responseBytes []byte,
+	err error,
+)
+
+type Client struct {
+	handlerID uint8
+	router    *Router
+	sender    common.AppSender
+}
+
+func (c *Client) AppRequestAny(
+	ctx context.Context,
+	appRequestBytes []byte,
+	onResponse AppResponseCallback,
+) error {
+	c.router.lock.RLock()
+	peers := maps.Keys(c.router.peers)
+	c.router.lock.RUnlock()
+
+	s := sampler.NewUniform()
+	s.Initialize(uint64(len(peers)))
+	drawn, err := s.Next()
+	if err != nil {
+		return err
+	}
+
+	nodeIDs := set.Set[ids.NodeID]{peers[drawn]: struct{}{}}
+	return c.AppRequest(ctx, nodeIDs, appRequestBytes, onResponse)
```

Comment:
We talked offline about why the bandwidth tracking is important. But I will say if we wanted to give the option for uniformly sampling nodes - we shouldn't flatten the map every time we do this... Iterating over a map returns random elements... so we can just use that instead.